enhance(breadcrumbs): show current and parent (not root) on mobile + add padding

Ahmed-Hakeem opened the pull request:
## Summary
Just a proposal from a user frequently using the app on mobile.
I found it better to show the previous nav item instead of the root for ex:
" Array> Array.prototype.slice() " instead of "References > Array.prototype.slice()"

Also, it handles breadcrumbs wrapped layout.

In addition to that, it shows the last two items only in small sizes.

Fixes #10385
(No longer fixes #8250,  #8627)

### Problem

Navbar shows 2 elements in small sizes ("root element", and "current page") and I just found the "root element" not needed by the user unlike the "parent of the current page"

Navbar not handling wrapped layouts.

Navbar shows 2 elements on a wide screen sizes where other items can be shown

### Solution

I suggest to replace the "root element" with the "parent of the current page"  so that we can provide the users with better experience in my opinion 

handle wrapped layouts by adding spaces between breadcrumbs items

show 2 elements only in small sizes 

## Screenshots

### Before

1- root and last items
![Screenshot 2024-01-25 105448](https://github.com/mdn/yari/assets/65634467/e97aefdd-1eb6-4430-8dae-acea7253ccb3)

2- wrapped layout with no space
![Screenshot 2024-01-27 064941](https://github.com/mdn/yari/assets/65634467/ceaa95e3-3990-4f90-9a21-8e5f64a721b8)

3- show 2 items even in wide screens
![Screenshot 2024-01-27 064819](https://github.com/mdn/yari/assets/65634467/f162d5fc-ecb3-4827-9716-75527b3edbf6)

### After

1-last two items
![image](https://github.com/mdn/yari/assets/65634467/f623dfe7-241a-4b74-837b-10bc3b9f3e72)

2-wrapped layout with space
![Screenshot 2024-01-27 064720](https://github.com/mdn/yari/assets/65634467/d8707c91-1bd7-4831-a7d6-680ae6e16c7c)

3-show 2 items in small screens only (size < 426px)
![Screenshot 2024-01-27 064751](https://github.com/mdn/yari/assets/65634467/5d813e2c-e74b-4e41-9406-0b6a4d97d476)

Ahmed-Hakeem committed:
show previous nav item in small sizes instead of the root nav item

caugner commented:
@Ahmed-Hakeem Thanks for opening this PR, but please make sure to follow and fill out the PR description template:

https://github.com/mdn/yari/blob/9838d6ca60bd72e0c32cff9f5ffa3546de3fc3d4/.github/PULL_REQUEST_TEMPLATE.md#L1-L14

Ahmed-Hakeem commented:
> @Ahmed-Hakeem Thanks for opening this PR, but please make sure to follow and fill out the PR description template:
> 
> https://github.com/mdn/yari/blob/9838d6ca60bd72e0c32cff9f5ffa3546de3fc3d4/.github/PULL_REQUEST_TEMPLATE.md#L1-L14

sure thing

wbamberg commented:
> Fixes https://github.com/mdn/yari/issues/8627 raised by @wbamberg by showing the parent instead of the root

This is better but it doesn't fix the issue. In my screenshot there's plenty of space to show all levels but Yari only shows one. In this PR, IIUC, in my example page we would see:

```
Guides > Making PWAs installable
```

...meaning I can't use the breadcrumbs to go to the top of the PWA docs.

Ahmed-Hakeem commented:
> > Fixes #8627 raised by @wbamberg by showing the parent instead of the root
> 
> This is better but it doesn't fix the issue. In my screenshot there's plenty of space to show all levels but Yari only shows one ancestor. In this PR, IIUC, in my example page we would see:
> 
> ```
> Guides > Making PWAs installable
> ```
> 
> ...meaning I can't use the breadcrumbs to go to the top of the PWA docs (or use the breadcrumbs to see the structure).

Yeah, I got it; I was about to mention that. It's not completely solving it.

I just didn't want to raise another issue with part of your concern. I found you saying, "At least that will enable me to get to the parent page," so that's why.

wbamberg commented:
Yes, totally fair - the only reason I'm being pedantic here is that adding "Fixes <bug number>" in the description means the bug will be auto-closed if/when the PR is merged, which IMO would not be appropriate.

Ahmed-Hakeem commented:
Should I raise another one, then ?

> Yes, totally fair - the only reason I'm being pedantic here is that adding "Fixes " in the description means the bug will be auto-closed if/when the PR is merged, which IMO would not be appropriate.

![image](https://github.com/mdn/yari/assets/65634467/9219b6d3-14e2-4efb-8588-acd924e265a6)

I have updated the description to clarify that

Ahmed-Hakeem committed:
show last two items in Breadcrumbs only in sizes less than 426px

Ahmed-Hakeem committed:
enhance wrapped layout of the breadcrumbs

Ahmed-Hakeem commented:
Now it completely fixes it @wbamberg  and fixes #8250

Ahmed-Hakeem committed:
Return auto hyphens style back to breadcrums items 

Return auto hyphens style back to breadcrums items since it was removed by mistake

caugner reviewed (changes requested):
Overall LGTM from a code perspective, just one nit regarding px vs rem, and just FYI this change needs approval from our product team, but I'll ping them.

Ahmed-Hakeem committed:
Stick to rems instead of pixels

Co-authored-by: Claas Augner <495429+caugner@users.noreply.github.com>

Ahmed-Hakeem commented on client/src/ui/molecules/breadcrumbs/index.scss:
```diff
@@ -64,7 +66,7 @@
   }
 
-  @media screen and (min-width: $screen-xl) {
+  @media screen and (min-width: $screen-sm) {
```
yeah, that's right ... it will display all li items on all screen sizes except xs ...... it will show only last 2 items on xs size

Ahmed-Hakeem replied:
remove it on xs size using display:"none" .... and display it as "inline-flex" on other sizes I mean

Ahmed-Hakeem commented:
> Overall LGTM from a code perspective, just one nit regarding px vs rem, and just FYI this change needs approval from our product team, but I'll ping them.

that's fine, waiting for it 

Thank you

caugner committed:
revert to initial version of PR

caugner commented on client/src/ui/molecules/breadcrumbs/index.scss:
```diff
@@ -12,10 +12,8 @@
-    // only show first and last on mobile
-    &:first-child,
-    &:last-child {
+    // only show last two items on small sizes (size < 426px)
```
```suggestion

    // only show last two items on small sizes (size < 426px)
```

caugner committed:
Update client/src/ui/molecules/breadcrumbs/index.scss

caugner committed:
fix(breadcrumbs): restore flex with correct padding and gap

Co-authored-by: Florian Dieminger <me@fiji-flo.de>

caugner commented on client/src/ui/molecules/breadcrumbs/index.scss:
```diff
@@ -6,16 +6,19 @@
-    // only show first and last on mobile
-    &:first-child,
-    &:last-child {
+    // only show last two items on small sizes (size < 426px)
```
```suggestion
    // only show last two items on mobile
```

caugner committed:
Update client/src/ui/molecules/breadcrumbs/index.scss

caugner reviewed (approved):
LGTU (@argl @fiji-flo @caugner), thanks! 🎉 

Note that we reduced the changes slightly to exclude the breakpoint change.

Ahmed-Hakeem commented:
Thank you @caugner for making changes on this one. I just want to mention that issue #8250 still solved with the new changes, also why we revert the breakpoint to be xl since this was what actually solves  #8627 and has no side effect

caugner commented:
> I just want to mention that issue #8250 still solved with the new changes,

Thank you, good catch, I have closed the issue again now. 👍 

> also why we revert the breakpoint to be xl since this was what actually solves #8627 and has no side effect

Good question. We need approval from our UX designer for UI changes like this one, but we're working on a redesign this year, so it doesn't make sense to change details like this, especially as it had unintended side-effects (note the locale line break and the heading - navigated to via the TOC - being slightly hidden by the header):
<img width="961" alt="image" src="https://github.com/mdn/yari/assets/495429/8087ce8c-9a5e-4dab-9648-e09c9bc3e47e">

Surely these side-effects can be fixed/mitigated, but we don't want to invest time into this right now, so we're not accepting PRs to fully resolve #8627 at this time.

Ahmed-Hakeem commented:
> > I just want to mention that issue #8250 still solved with the new changes,
> 
> Thank you, good catch, I have closed the issue again now. 👍
> 
> > also why we revert the breakpoint to be xl since this was what actually solves #8627 and has no side effect
> 
> Good question. We need approval from our UX designer for UI changes like this one, but we're working on a redesign this year, so it doesn't make sense to change details like this, especially as it had unintended side-effects (note the locale line break and the heading - navigated to via the TOC - being slightly hidden by the header): <img alt="image" width="961" src="https://private-user-images.githubusercontent.com/495429/320009886-8087ce8c-9a5e-4dab-9648-e09c9bc3e47e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI4NTc1ODQsIm5iZiI6MTcxMjg1NzI4NCwicGF0aCI6Ii80OTU0MjkvMzIwMDA5ODg2LTgwODdjZThjLTlhNWUtNGRhYi05NjQ4LWUwOWM5YmMzZTQ3ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxMVQxNzQxMjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMmNmMmNiZWMwYjdlYzg1NTdkZWE2OWQxOWY3YjUwMTNlNzJhYWM3ZDM1YzA3MTdhYzYzZjNiNzA0YzA4YzJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Ig83d2Inq5iFeAkIKaj0oUMVMxuHizQKY1zZdkWIYgA">
> 
> Surely these side-effects can be fixed/mitigated, but we don't want to invest time into this right now, so we're not accepting PRs to fully resolve #8627 at this time.

Thank you for your clarification : ) …. excited to see the new design ❤️.  Keep up the good work. 
